### PR TITLE
refactor: bank transaction counts

### PR DESCRIPTION
**Pull request by jstarry**
#### Problem
Hard to follow how transaction counts are used throughout the runtime

#### Summary of Changes
- Extract a few new structs for tracking transaction counts: `ExecuteAndCommitTransactionsCounts`  and `ProcessTransactionsCounts`
- Switch to `u64` from `usize` for a number of counts

Fixes #

### CODE REVIEW

File: core/src/banking_stage/leader_slot_metrics.rs [hunk 1]
```diff
@@ -19,33 +20,22 @@ use {
...
-    pub committed_transactions_count: usize,
-
-    // Total number of transactions that made it into the block where the transactions
-    // output from execution was success/no error.
-    pub committed_transactions_with_successful_result_count: usize,
-
-    // All transactions that were executed but then failed record because the
-    // slot ended
-    pub failed_commit_count: usize,
+    // Total transaction counts tracked for reporting `LeaderSlotMetrics`. See
```
Comment (apfitzge) [inline core/src/banking_stage/leader_slot_metrics.rs]: consistent with other comments so fine to leave as is, but why would all the comments on theese fields not be doc-comments?

File: core/src/banking_stage/leader_slot_metrics.rs [hunk 2]
```diff
@@ -60,6 +50,48 @@ pub(crate) struct ProcessTransactionsSummary {
     pub max_prioritization_fees: u64,
 }
 
+#[derive(Debug, Default, PartialEq)]
+pub struct ProcessTransactionsCounts {
+    // Total number of transactions that were passed as candidates for execution
```
Comment (apfitzge) [inline core/src/banking_stage/leader_slot_metrics.rs]: /// doc-comments here too?

File: core/src/banking_stage/consumer.rs [hunk 3]
```diff
@@ -46,21 +48,16 @@ pub const TARGET_NUM_TRANSACTIONS_PER_BATCH: usize = 64;
 
 pub struct ProcessTransactionBatchOutput {
     // The number of transactions filtered out by the cost model
-    pub(crate) cost_model_throttled_transactions_count: usize,
```
Comment (apfitzge) [inline core/src/banking_stage/consumer.rs:49]: noticed this in a few places. Why use u64 for counts over usize?

**Review by apfitzge — DISMISSED** [review]
only nits, changes look fine to me. think the new struct(s) help readability

File: core/src/banking_stage/consumer.rs (hunk 3 above)
Comment (jstarry) [reply core/src/banking_stage/consumer.rs:49]: Because they are all eventually converted to u64 for metrics reporting or for updating bank stats

Comment (ksolana) [reply core/src/banking_stage/consumer.rs:49]: As usize is u64 on 64 bit machines, this change should have no effect but tbh usize is confusing.

**Review by apfitzge — APPROVED** [review]
